### PR TITLE
Build docs transport package in CI

### PR DESCRIPTION
**Pull request by carlossanlop**
The recent addition of slngen.ps1 only considers cs and vb project files, but not proj files. This causes the new docs transport package from getting built.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/extensions/pull/4562)

### CODE REVIEW

File: eng/pipelines/templates/BuildAndTest.yml [hunk 1]
```diff
@@ -47,6 +47,7 @@ steps:
 
   - pwsh: |
         $(Build.SourcesDirectory)/scripts/Slngen.ps1 -All -NoLaunch
+        $(Build.SourcesDirectory)/.dotnet/dotnet sln add --project $(Build.SourcesDirectory)/src/Packages/Microsoft.Internal.Extensions.DotNetApiDocs.Transport/Microsoft.Internal.Extensions.DotNetApiDocs.Transport.proj" />
```
Comment (RussKie) [inline eng/pipelines/templates/BuildAndTest.yml:50]:
```suggestion
        $(Build.SourcesDirectory)/.dotnet/dotnet sln add --project $(Build.SourcesDirectory)/src/Packages/Microsoft.Internal.Extensions.DotNetApiDocs.Transport/Microsoft.Internal.Extensions.DotNetApiDocs.Transport.proj
```

File: eng/pipelines/templates/BuildAndTest.yml [hunk 2]
```diff
@@ -47,6 +47,7 @@ steps:
 
   - pwsh: |
         $(Build.SourcesDirectory)/scripts/Slngen.ps1 -All -NoLaunch
+        $(Build.SourcesDirectory)/.dotnet/dotnet sln add --project $(Build.SourcesDirectory)/src/Packages/Microsoft.Internal.Extensions.DotNetApiDocs.Transport/Microsoft.Internal.Extensions.DotNetApiDocs.Transport.proj
```
Comment (RussKie) [inline eng/pipelines/templates/BuildAndTest.yml:50]: @carlossanlop did you try this command locally?

Comment (carlossanlop) [reply eng/pipelines/templates/BuildAndTest.yml:50]:
```suggestion
        $(Build.SourcesDirectory)/.dotnet/dotnet sln add $(Build.SourcesDirectory)/src/Packages/Microsoft.Internal.Extensions.DotNetApiDocs.Transport/Microsoft.Internal.Extensions.DotNetApiDocs.Transport.proj
```

**Comment by carlossanlop** [comment]
Apparently slns do not like projs:
```
Project 'D:\a\_work\1\s\src\Packages\Microsoft.Internal.Extensions.DotNetApiDocs.Transport\Microsoft.Internal.Extensions.DotNetApiDocs.Transport.proj' has an unknown project type and cannot be added to the solution file. Contact your SDK provider for support.
Finishing: Create solution
```

File: src/Packages/Microsoft.Internal.Extensions.DotNetApiDocs.Transport/Microsoft.Internal.Extensions.DotNetApiDocs.Transport.proj [hunk 3]
```diff
@@ -25,11 +25,8 @@
...
     <ProjectReference PrivateAssets="all"
       Private="false"
-      Include="$(SrcLibrariesDir)\*\*.*proj;
-               $(SrcLibrariesDir)\..\ToBeMoved\*\*.*proj;
-               $(SrcLibrariesDir)\..\ToBeRemoved\*\*.*proj"
-      Exclude="$(MSBuildProjectFullPath)"
-                      />
+      Include="$(SrcLibrariesDir)\*\*.*proj"
+      Exclude="$(MSBuildProjectFullPath)" />
   </ItemGroup>
```
Comment (carlossanlop) [inline src/Packages/Microsoft.Internal.Extensions.DotNetApiDocs.Transport/Microsoft.Internal.Extensions.DotNetApiDocs.Transport.proj:30]: The changes in this file address the post-merge feedback from @RussKie in the previous PR: https://github.com/dotnet/extensions/pull/4254#pullrequestreview-1665750137

**Review by RussKie — APPROVED** [review]

**Pull request merged**